syntax = "proto3";
package protos.connectors.assets;

import "google/protobuf/wrappers.proto";
import "protos/base.proto";
import "protos/connectors/connector.proto";

message ClickhouseDatabaseAssetModel {
  google.protobuf.StringValue database = 1;
}

message ClickhouseDatabaseAssetOptions {
  repeated string databases = 1;
}

message ClickhouseAssetModel {
  google.protobuf.UInt64Value id = 1;
  Source connector_type = 2;
  SourceModelType type = 3;
  sfixed64 last_updated = 4;

  oneof asset {
    ClickhouseDatabaseAssetModel clickhouse_database = 5;
  }
}

message ClickhouseAssets {
  repeated ClickhouseAssetModel assets = 1;
}
